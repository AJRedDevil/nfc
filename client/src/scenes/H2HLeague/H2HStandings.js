// npm packages
import React from 'react';

// our packages
import Animation from '../../components/animations';
import Table from '../../components/Table';
import H2HStandingsPropTypes from './PropTypes';

const getBody = standings =>
  standings.map(item => [
    item.rank,
    item.entry_name,
    item.player_name,
    item.matches_won,
    item.matches_drawn,
    item.matches_lost,
    item.points_for,
    item.points_total,
  ]);

const H2HStandings = props => (
  <Animation.Fade in>
    <div>
      <Table body={getBody(props.standings)} head={props.head} />
    </div>
  </Animation.Fade>
);
H2HStandings.propTypes = {
  head: H2HStandingsPropTypes.head,
  standings: H2HStandingsPropTypes.body,
};

export default H2HStandings;
