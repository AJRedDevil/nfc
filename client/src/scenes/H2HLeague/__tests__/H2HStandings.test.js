// npm packages
import React from 'react';
import renderer from 'react-test-renderer';

// our packages
import H2HStandings, {getBody} from '../H2HStandings';

const standings = [
  {
    rank: 1,
    entry_name: 'test',
    player_name: 'test_1',
    matches_won: 1,
    matches_drawn: 1,
    matches_lost: 1,
    points_for: 1,
    points_total: 100,
  },
  {
    rank: 2,
    entry_name: 'test',
    player_name: 'test_2',
    matches_won: 2,
    matches_drawn: 2,
    matches_lost: 2,
    points_for: 2,
    points_total: 200,
  },
];

describe('# H2HStandings', () => {
  test('get correct body from standings', () => {
    const response = getBody(standings);
    expect(response).toEqual([
      [1, 'test', 'test_1', 1, 1, 1, 1, 100],
      [2, 'test', 'test_2', 2, 2, 2, 2, 200],
    ]);
  });

  test('H2HStandings wrapper', () => {
    const head = [
      {text: 'Rank'},
      {text: 'Team'},
      {text: 'Player'},
      {text: 'Won'},
      {text: 'Drawn'},
      {text: 'Lost'},
      {text: 'For'},
      {text: 'Total'},
    ];
    const wrapper = renderer
      .create(<H2HStandings head={head} standings={standings} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
