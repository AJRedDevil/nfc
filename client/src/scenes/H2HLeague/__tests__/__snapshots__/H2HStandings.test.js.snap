// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# H2HStandings H2HStandings wrapper 1`] = `
<div>
  <table
    className="table is-fullwidth is-striped is-narrow is-hoverable"
  >
    <thead>
      <tr>
        <th>
          Rank
        </th>
        <th>
          Team
        </th>
        <th>
          Player
        </th>
        <th>
          Won
        </th>
        <th>
          Drawn
        </th>
        <th>
          Lost
        </th>
        <th>
          For
        </th>
        <th>
          Total
        </th>
      </tr>
    </thead>
    <tfoot>
      <tr />
    </tfoot>
    <tbody>
      <tr>
        <th>
          1
        </th>
        <td>
          test
        </td>
        <td>
          test_1
        </td>
        <td>
          1
        </td>
        <td>
          1
        </td>
        <td>
          1
        </td>
        <td>
          1
        </td>
        <td>
          100
        </td>
      </tr>
      <tr>
        <th>
          2
        </th>
        <td>
          test
        </td>
        <td>
          test_2
        </td>
        <td>
          2
        </td>
        <td>
          2
        </td>
        <td>
          2
        </td>
        <td>
          2
        </td>
        <td>
          200
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
